course05/homework01/joel/string.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <ostream>
+#include <string>
+
+class String
+{
+public:
+	String(const char s[]) : mSize(std::strlen(s)), mContents(std::make_unique<char[]>(mSize))
+	{
+		std::strcpy(mContents.get(), s);
+	}
+	String() : String("") {}
+	String(const std::string s) : String(s.c_str()) {}
```

Comment:
This is copying (extra work, allocation) the string , it should take a const ref